pkg/validations/createvalidations/gitops_test.go

```diff
@@ -315,3 +330,172 @@ func TestValidateGitopsForSelfManagedCluster(t *testing.T) {
+			})
+			k, ctx, cluster, _ := validations.NewKubectl(t)
+			cluster.Name = "management-cluster"
+
+			err := createvalidations.ValidateGitOps(ctx, k, cluster, clusterSpec)
+			if !reflect.DeepEqual(err, tc.wantErr) {
+				t.Errorf("%v got = %v, \nwant %v", tc.name, err, tc.wantErr)
+			}
+		})
+	}
+}
+
+func TestValidateGitProviderAuthPrivateKeyForWorkloadCluster(t *testing.T) {
+	saveContext(t, emptyVar, testEnvVar)
+	tests := []struct {
+		name    string
+		wantErr error
+		git     v1alpha1.GitProviderConfig
+	}{
+		{
+			name:    "Empty password",
+			wantErr: fmt.Errorf("private key file does not exist at %s or is empty", testEnvVar),
+			git: v1alpha1.GitProviderConfig{
+				Username:      "testUser",
+				RepositoryUrl: "testRepo",
+			},
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(tt *testing.T) {
+			defaultFlux := &v1alpha1.FluxConfig{
+				Spec: v1alpha1.FluxConfigSpec{
+					Git: &tc.git,
+				},
+			}
+			clusterSpec := test.NewClusterSpec(func(s *cluster.Spec) {
+				s.Cluster.Name = testclustername
+				s.Cluster.Spec.GitOpsRef = &v1alpha1.Ref{
+					Kind: v1alpha1.GitOpsConfigKind,
+					Name: "gitopstest",
+				}
+				s.FluxConfig = defaultFlux
+				s.Cluster.SetManagedBy("management-cluster")
+			})
+			k, ctx, cluster, _ := validations.NewKubectl(t)
+			cluster.Name = "management-cluster"
+
+			err := createvalidations.ValidateGitOps(ctx, k, cluster, clusterSpec)
+			if !reflect.DeepEqual(err, tc.wantErr) {
+				t.Errorf("%v got = %v, \nwant %v", tc.name, err, tc.wantErr)
+			}
+		})
+	}
+}
+
+func TestValidateGitProviderAuthForWorkloadCluster(t *testing.T) {
+	saveContext(t, emptyVar, emptyVar)
+	tests := []struct {
+		name    string
+		wantErr error
+		git     v1alpha1.GitProviderConfig
+	}{
+		{
+			name:    "Empty password and private key",
+			wantErr: fmt.Errorf("for git provider in the Flux config either set a password using %s env var or path to private key file using %s", eksaGitPasswordTokenEnv, eksaGitPrivateKeyTokenEnv),
+			git: v1alpha1.GitProviderConfig{
+				Username:      "testUser",
+				RepositoryUrl: "testRepo",
+			},
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(tt *testing.T) {
+			defaultFlux := &v1alpha1.FluxConfig{
+				Spec: v1alpha1.FluxConfigSpec{
+					Git: &tc.git,
+				},
+			}
+			clusterSpec := test.NewClusterSpec(func(s *cluster.Spec) {
+				s.Cluster.Name = testclustername
+				s.Cluster.Spec.GitOpsRef = &v1alpha1.Ref{
+					Kind: v1alpha1.GitOpsConfigKind,
+					Name: "gitopstest",
+				}
+				s.FluxConfig = defaultFlux
+				s.Cluster.SetManagedBy("management-cluster")
+			})
+			k, ctx, cluster, _ := validations.NewKubectl(t)
+			cluster.Name = "management-cluster"
+
+			err := createvalidations.ValidateGitOps(ctx, k, cluster, clusterSpec)
+			if !reflect.DeepEqual(err, tc.wantErr) {
+				t.Errorf("%v got = %v, \nwant %v", tc.name, err, tc.wantErr)
+			}
+		})
+	}
+}
+
+func TestValidateNoGitProviderAuthForWorkloadCluster(t *testing.T) {
```

Comment:
Can we rename these methods to follow the pattern `Test[entity-name][tested-method-name][scenario]`?